src/java.base/share/classes/sun/reflect/annotation/AnnotationInvocationHandler.java

```diff
@@ -189,6 +191,10 @@ else if (type == Long.class)
                 return  toSourceString((long) value);
             else if (type == Byte.class)
                 return  toSourceString((byte) value);
+            else if (value instanceof Enum<?>)
+                // Predicate above covers enum constants, including
+                // those with specialized class bodies.
+                return toSourceString((Enum<?>) value);
```

Comment:
Hello Joe, would it be better to use the new syntax for `instanceof` here to avoid the subsequent cast?

```
else if (value instance of Enum<?> v) 
  ....
  return toSourceString(v);
```

Reply:
> Hello Joe, would it be better to use the new syntax for `instanceof` here to avoid the subsequent cast?
> 
> ```
> else if (value instanceof Enum<?> v) 
>   ....
>   return toSourceString(v);
> ```

Fair point; updated in subsequent push. Thanks.

Reply:
Thank you for this change, Joe. Looks fine to me.